web/src/copy_and_paste.ts

```diff
@@ -402,6 +402,23 @@ export function paste_handler_converter(paste_html: string): string {
         .parseFromString(paste_html, "text/html")
         .querySelector("body");
     assert(copied_html_fragment !== null);
+
+    const para_elements = copied_html_fragment.querySelectorAll("p");
+
+    for (const element of para_elements) {
+        const children = [...element.childNodes];
+        for (const child of children) {
+            // Check if the current node is a text node
+            if (child.nodeType === Node.TEXT_NODE) {
+                // Create a new span for the text node
+                const span = document.createElement("span");
+                span.classList.add("intermediate-katex-text-node");
+                span.textContent = child.textContent;
+                child.replaceWith(span);
+            }
+        }
+    }
+
     const copied_within_single_element = within_single_element(copied_html_fragment);
     const outer_elements_to_retain = ["PRE", "UL", "OL", "A", "CODE"];
     // If the entire selection copied is within a single HTML element (like an
```

Comment:
How did you decide whether the spans for text nodes need to happen before or after this?